pkg/controller/elasticsearch/validation/warnings.go

```diff
@@ -26,10 +37,35 @@ func noUnsupportedSettings(es esv1.Elasticsearch) field.ErrorList {
 			errs = append(errs, field.Invalid(field.NewPath("spec").Child("nodeSets").Index(i).Child("config"), es.Spec.NodeSets[i].Config, cfgInvalidMsg))
 			continue
 		}
-		unsupported := config.HasKeys(esv1.UnsupportedSettings)
-		for _, setting := range unsupported {
-			errs = append(errs, field.Forbidden(field.NewPath("spec").Child("nodeSets").Index(i).Child("config").Child(setting), unsupportedConfigErrMsg))
-		}
+		newErrs := check(config, i)
+		errs = append(errs, newErrs...)
+	}
+	return errs
+}
+
+func validateSettings(config *common.CanonicalConfig, index int) field.ErrorList {
+	var errs field.ErrorList
+	unsupported := config.HasKeys(esv1.UnsupportedSettings)
+	for _, setting := range unsupported {
+		errs = append(errs, field.Forbidden(field.NewPath("spec").Child("nodeSets").Index(index).Child("config").Child(setting), unsupportedConfigErrMsg))
+	}
+	return errs
+}
+
+func validateClientAuthentication(config *common.CanonicalConfig, index int) field.ErrorList {
+	type ClientAuthSetting struct {
+		Value string `config:"xpack.security.http.ssl.client_authentication"`
```

Comment:
Let me know if there is an easier / generic way to access a setting given the key as string, so we don't need to define a struct for it.

Reply:
Does this resolve path separators correctly ? I.e. `xpack.security.http...` I don't remember.

Reply:
Looks like it works as expected! I added the wrapper and a few basic test cases.